util-flywayinstaller-cli/src/main/java/com/github/cafapi/util/flywayinstaller/Application.java

```diff
@@ -0,0 +1,121 @@
+ * limitations under the License.
+ */
+package com.github.cafapi.util.flywayinstaller;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.LoggerContext;
+import java.sql.SQLException;
+import java.util.concurrent.Callable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "util-flywayinstaller",
+    mixinStandardHelpOptions = true,
+    exitCodeListHeading = "%nExit Codes:%n",
+    exitCodeList = {" 0:Successful program execution.",
+                    " 1:Invalid input: An invalid parameter was specified.",
+                    " 2:Execution exception: An exception occurred while executing the migrations."})
+public final class Application implements Callable<Integer>
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(Application.class);
+
+    private Application()
+    {
+    }
+
+    @CommandLine.Option(
+        names = {"-db.host"},
+        paramLabel = "<dbHost>",
+        required = true,
+        description = "Specifies the database host name. "
+        + "e.g. localhost"
+    )
+    private String dbHost;
+
+    @CommandLine.Option(
+        names = {"-db.port"},
+        paramLabel = "<dbPort>",
+        required = true,
+        description = "Specifies the database port to be used. "
+        + "e.g. 5432"
+    )
+    private int dbPort;
+
+    @CommandLine.Option(
+        names = {"-db.user"},
+        paramLabel = "<username>",
+        required = true,
+        description = "Specifies the username to access the database."
+    )
+    private String username;
+
+    @CommandLine.Option(
+        names = {"-db.pass"},
+        paramLabel = "<password>",
+        required = true,
+        description = "Specifies the password to access the database."
+    )
+    private String password;
+
+    @CommandLine.Option(
+        names = {"-db.name"},
+        paramLabel = "<dbName>",
+        required = true,
+        description = "Specifies the name of the database to be created or updated."
+    )
+    private String dbName;
+
+    @CommandLine.Option(
+        names = {"-log"},
+        paramLabel = "<logLevel>",
+        description = "Specifies the logging level of the installer. Can be DEBUG, INFO, WARNING, ERROR or OFF. Must be Uppercase"
+    )
+    private LogLevel logLevel;
+
+    public static void main(final String[] args)
+    {
+        final int exitCode = new CommandLine(new Application()).execute(args);
+        System.exit(exitCode);
+    }
+
+    @Override
+    public Integer call()
+    {
+        if (logLevel != null) {
+            setLogLevel(logLevel);
+        }
+
+        try {
+            Migrator.migrate(dbHost, dbPort, dbName, username, password);
+        } catch (final SQLException ex) {
+            LOGGER.error("Issue while trying to perform the migration.", ex);
+            return 1;
+        } catch (final Exception ex) {
```

Comment:
What type of exception are you trying to catch here?

Reply:
2 types of exceptions mainly: authentication, then anything else that would classify as migration issues. I adapted the code to reflect it. Picocli takes already care of the invalid input exceptions